### PR TITLE
Add request to middleware Preprocess for unary interceptor use

### PR DESCRIPTION
**Pull request by lumjjb**
Signed-off-by: Brandon Lum <lumjjb@gmail.com>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md? 🤞 
- [x] Proper tests/regressions included? (interceptor_test.go modified to check passing of request body)
- [x] Documentation updated? (code in-line documentation)

**Affected functionality**

This PR adds a field to the middleware `Preprocess` function allowing passing of the request object for Unary gRPC calls. This will allow future enhancement to make more granular actions for `Preprocess` functions, i.e. for authorization to have finer grained policies (#1975) based on the body of the request. 

**Description of change**

Add the `req interface{}` field for `middleware.Preprocess`. The request is passed on via the UnaryInterceptor, but for StreamInterceptors the `req` value is `nil`. 

**Which issue this PR fixes**

NA

### CODE REVIEW

File: pkg/common/api/middleware/middleware.go [hunk 1]
```diff
@@ -4,15 +4,17 @@ import (
...
 
 type Middleware interface {
 	// Preprocess is invoked before the gRPC handler is called. It returns a
 	// (possibly modified) context that is passed into the handler, which
 	// should either be the context passed into the function or one derived
 	// from it. If the function returns an error, the gRPC method fails.
-	Preprocess(ctx context.Context, fullMethod string) (context.Context, error)
+	// req passes the request object for unary interceptors and nil for
+	// stream interceptors
+	Preprocess(ctx context.Context, req interface{}, fullMethod string) (context.Context, error)
```
Comment (azdagron) [inline pkg/common/api/middleware/middleware.go:17]: The nittiest of all nitpicks :) How would you feel about adding the `req` parameter after `fullMethod`? That has the benefit of keeping the first two arguments the same between `Preprocess` and `Postprocess`.

**Review by azdagron — COMMENTED** [review]
Thanks, @lumjjb. I just have one small nitpick if you don't mind :)

File: pkg/common/api/middleware/middleware.go (hunk 1 above)
Comment (lumjjb) [reply pkg/common/api/middleware/middleware.go]: of course :). Done!

**Review by azdagron — APPROVED** [review]
\o/

**Pull request merged**